tiledb/sm/query/query.cc

```diff
@@ -88,9 +90,9 @@ Query::Query(StorageManager* storage_manager, Array* array, URI fragment_uri)
   assert(st.ok());
 
   if (type_ == QueryType::WRITE) {
-    subarray_ = Subarray(array, stats_, logger_);
+    subarray_ = Subarray(array.get(), stats_, logger_);
```

Comment:
Can just use `array_` here. No need to call `get()`; the object member is already initialized by the time the constructor body runs.